zebra/rt_netlink.c

```diff
@@ -1340,6 +1340,8 @@ static void _netlink_route_build_multipath(const char *routedesc, int bytelen,
 			      bytelen);
 		rtnh->rtnh_len += sizeof(struct rtattr) + bytelen;
 		rtnh->rtnh_ifindex = nexthop->ifindex;
+		if (nexthop->weight)
```

Comment:
Similar code is present at the end of this function as well. May be take this out to some common place in this function ?

Reply:
Since that is unrelated to this commit, I didn't handle as part of this change. This commit just sets the rtnh_hops parameter which was missing in this case.